src/corehost/common/trace.cpp

```diff
@@ -166,3 +178,19 @@ void trace::flush()
     pal::err_flush();
     pal::out_flush();
 }
+
+trace::error_writer_fn trace::set_error_writer(trace::error_writer_fn error_writer)
+{
+    std::lock_guard<std::mutex> lock(g_trace_mutex);
+
+    error_writer_fn previous_writer = g_error_writer;
+    g_error_writer = error_writer;
+    return previous_writer;
+}
+
+trace::error_writer_fn trace::get_error_writer()
+{
+    std::lock_guard<std::mutex> lock(g_trace_mutex);
```

Comment:
Since these are thread local I do not think the lock is necessary/appropriate

Reply:
Good point!